### PR TITLE
feat(TDP-1463 TDP-2195): scope dataset and deduplication

### PR DESCRIPTION
**Pull request by tlnd-cgall**
https://jira.talendforge.org/browse/TDP-1463
https://jira.talendforge.org/browse/TDP-2195

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/767/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/767/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/768/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/768/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/861/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/861/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/875/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/875/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/889/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/889/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/890/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/890/)

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+@Action(AbstractActionMetadata.ACTION_BEAN_PREFIX + Deduplicate.DEDUPLICATION_ACTION_NAME)
+public class Deduplicate extends AbstractActionMetadata implements DataSetAction {
+
+    /**
+     * The action code name.
+     */
+    public static final String DEDUPLICATION_ACTION_NAME = "deduplication";
+
+    /** Hashes name. */
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java]: Explain the purpose of this string (like key to store in context hashes....)

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    public static final String HASHES_NAME = "hashes";
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java]: change it to private

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    @Override
+    public Set<Behavior> getBehavior() {
+        return EnumSet.of(Behavior.FORBID_DISTRIBUTED, Behavior.VALUES_DELETE_ROWS);
+    }
+
+    @Override
+    public void compile(ActionContext actionContext) {
+        final Set<String> hashes = new HashSet<>();
+        actionContext.get(HASHES_NAME, p -> hashes);
+        super.compile(actionContext);
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/duplication/Deduplicate.java]: Generally, super. is the first line of the function. Is it on purpose this is not the case here?

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/892/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/892/)

File: dataprep-webapp/yarn.lock [hunk 4]
```diff
@@ -6627,6 +6627,10 @@ yargs@~3.10.0:
     decamelize "^1.0.0"
     window-size "0.1.0"
 
+yarn@^1.2.1:
+  version "1.3.1"
+  resolved "https://registry.yarnpkg.com/yarn/-/yarn-1.3.1.tgz#ed4840e1985baa4f1e3eb9667d467a2888f498d5"
+
```
Comment (fabballe) [inline dataprep-webapp/yarn.lock]: See comments on package.json

File: dataprep-webapp/package.json [hunk 5]
```diff
@@ -62,7 +62,8 @@
     "react-css-transition": "^0.7.4",
     "react-dom": "^15.4.2",
     "sunchoke": "git+https://github.com/Talend/sunchoke.git#2.0.0",
-    "uuid": "^2.0.3"
+    "uuid": "^2.0.3",
+    "yarn": "^1.2.1"
```
[the remark this answers is not in the document]
Comment (tlnd-cgall) [reply dataprep-webapp/package.json]: Dont know, I don't using yarn but npm

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/902/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/902/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/903/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/903/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/906/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/906/)

**Pull request merged**